Add recipe for notmuch-bookmarks

publicimageltd opened the pull request:
### Brief summary of what the package does

Add bookmark handling for notmuch buffers.

### Direct link to the package repository

https://github.com/publicimageltd/notmuch-bookmarks

### Your association with the package

Author

### Relevant communications with the upstream package maintainer

*None needed*

### Checklist

Please confirm with `x`:

- [ ] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).

Do I have to add a license file (in addition to the declarations in the source file)? If yes, could you point out which one? 

- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback

- [x] My elisp byte-compiles cleanly

- [x] `M-x checkdoc` is happy with my docstrings

- [ ] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

No, I have not. I had enough troubles to find out HOW to contribute (forking melpa etc.). I could not find out how to test the installation of the package in a clean vanilla emacs environment. In my own environment, the package is loaded anyways. So I did not figure out how to easily simulate a clean install in an environment yet unware of my package.

- [ ] I have confirmed some of these without doing them

### FURTHER NOTE

I am a layman, and I had quite some trouble figuring out how to contribute. The guidelines in the melpa package are a bit too short. In particular, the 'testing' part remains obscure to me. I am glad about any further points. I would also add to the documentation once I have better understood the process; yet not being a native speaker, I am reluctant.

riscy commented:
Don't worry, a lot of people have trouble contributing for the first time.  I ran your package through some tests:

### notmuch-bookmarks.el

<kbd>M-x byte-compile-file</kbd> (using Emacs 26.3):
```
notmuch-bookmarks.el:47:1:Error: Cannot open load file: No such file or directory, notmuch
```
This is because you also need to include `notmuch` on your `;; Package-Requires: ` line.  Add that in and ping me and I'll rerun.

<kbd>M-x checkdoc</kbd> (using version 0.6.1):
No issues! 💯

<kbd>M-x package-lint-current-buffer</kbd> (using version 20191124.132):
No issues! 💯

Other static checks:
- notmuch-bookmarks.el#L6: Prefer `;; Version` instead of `;; Package-Version` (MELPA automatically adds `Package-Version`)
- notmuch-bookmarks.el#L289: Advice functions should be sharp-quoted; use `#'` (optional)
- notmuch-bookmarks.el#L76: It's safer to sharp-quote function names; use `#'` (optional)

As for your license, you do meet the requirement of having a GPL-compatible license.  If you create a "LICENSE" or "COPYING" file in your repo with a copy of the license, then GitHub will automatically detect this and give you a little GPL icon, and the GitHub API will correctly identify your license (so it's a bonus).  But having the license in your elisp file(s) is sufficient.

alphapapa commented:
> No, I have not. I had enough troubles to find out HOW to contribute (forking melpa etc.). I could not find out how to test the installation of the package in a clean vanilla emacs environment. In my own environment, the package is loaded anyways. So I did not figure out how to easily simulate a clean install in an environment yet unware of my package.

This is explained in `CONTRIBUTING.org`: https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org#test-your-recipe  See the part about the sandbox.

publicimageltd commented:
Thanks for the fast feedback. I managed to test the package in a sandboxed emacs (thanks for the pointer) and added some changes. Only open question left, I hope, is the "versioning" stuff. Since there are new changes in the repo now, do we have to move one version up and adjust the recipe?

riscy commented:
You don't have to bump your version, especially since this hasn't been merged yet.

MELPA will create a package-version that gets distributed to everyone who installs your package using MELPA (e.g. `;; Package-Version: 20171119.646`).  You can bump your `;; Version` if you ever decide to make a stable release by tagging your repo (which will put you in [MELPA Stable](https://github.com/melpa/melpa#melpa-stable)).

riscy commented:
I should be more specific -- your `;; Version:` is yours to use how you see fit.  Some people pin it at 0 and let MELPA handle the package-versioning for them.

riscy commented:
I think the byte-compile output I added earlier is still relevant.  Feel free to ping me when this is ready for another pass!

publicimageltd commented:
I cannot reproduce the problem with the latest build. Byte 
compiling the file with a vanilla emacs (sandbox) only yields two 
warning about unused lexical arguments. I am also using emacs 
version 26.3. I had updated the package with `make 
recipes/notmuch-bookmarks` and then called `make sandbox 
INSTALL=notmuch-bookmarks`. In this plain emacs, I byte-compiled 
the file.

Chris Rayner <notifications@github.com> writes:

> I think the byte-compile output I added earlier is still 
> relevant. Feel free to ping me when this is ready for another 
> pass! 
> 
> — You are receiving this because you authored the thread.  Reply 
> to this email directly, view it on GitHub, or unsubscribe. 

-- 
http://www.joergvolbers.de
https://fu-berlin.academia.edu/jvolbers

publicimageltd commented:
I merged the changes, everything should be fine now. I do not know whether there is something I have to do now. I'll patiently wait, no hurry; no need even to answer; I just wanted to drop this message to make sure that nothing is missing.

riscy commented:
Apologies for how long this turnaround was.  This looks ready to go to me.  And, welcome to MELPA!

publicimageltd commented:
Thanks for your work, I appreciate it!
